pkg/alertmanager/multitenant.go

```diff
@@ -366,70 +368,55 @@ func (am *MultitenantAlertmanager) setConfig(userID string, config configs.Confi
 		}
 	}
 
-	if config.AlertmanagerConfig == "" {
+	level.Debug(util.Logger).Log("msg", "MultitenantAlertmanager: setting config", "user", cfg.User)
+
+	if cfg.RawConfig == "" {
 		if am.fallbackConfig == "" {
-			return fmt.Errorf("blank Alertmanager configuration for %v", userID)
+			return fmt.Errorf("blank Alertmanager configuration for %v", cfg.User)
 		}
-		level.Info(util.Logger).Log("msg", "blank Alertmanager configuration; using fallback", "user_id", userID)
-		amConfig, err = amconfig.Load(am.fallbackConfig)
+		level.Info(util.Logger).Log("msg", "blank Alertmanager configuration; using fallback", "user_id", cfg.User)
+		userAmConfig, err = amconfig.Load(am.fallbackConfig)
 		if err != nil {
-			return fmt.Errorf("unable to load fallback configuration for %v: %v", userID, err)
+			return fmt.Errorf("unable to load fallback configuration for %v: %v", cfg.User, err)
 		}
 	} else {
-		amConfig, err = alertmanagerConfigFromConfig(config)
+		userAmConfig, err = amconfig.Load(cfg.RawConfig)
 		if err != nil && hasExisting {
 			// XXX: This means that if a user has a working configuration and
 			// they submit a broken one, we'll keep processing the last known
 			// working configuration, and they'll never know.
 			// TODO: Provide a way of communicating this to the user and for removing
 			// Alertmanager instances.
-			return fmt.Errorf("invalid Cortex configuration for %v: %v", userID, err)
+			return fmt.Errorf("invalid Cortex configuration for %v: %v", cfg.User, err)
 		}
 	}
 
-	if amConfig, err = am.transformConfig(userID, amConfig); err != nil {
+	if userAmConfig, err = am.transformConfig(cfg.User, userAmConfig); err != nil {
 		return err
 	}
 
 	// If no Alertmanager instance exists for this user yet, start one.
 	if !hasExisting {
-		newAM, err := am.newAlertmanager(userID, amConfig)
+		level.Debug(util.Logger).Log("msg", "MultitenantAlertmanager: initializing new alertmanager tenant", "user", cfg.User)
```

Comment:
I'm seeing several logs calling it "alertmanager tenant". I personally find it a bit unclear. Maybe checking all the logs in this file and referring to it as the "per-tenant alertmanager" may be more clear? _I'm not a native english, so feel free to ignore this comment if you think the current one is good._

Reply:
I like `per-tenant alertmanager`. I'll update it.